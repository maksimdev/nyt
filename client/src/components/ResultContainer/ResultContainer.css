.listContainer {
    list-style-type: none;
}

.listItem {
    margin-bottom: 20px; 
}

.listItem .headline {
    font-size: 1.5em;
}

.listItem .source {
    font-size: 0.7em;
}