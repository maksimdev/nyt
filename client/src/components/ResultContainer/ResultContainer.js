import React from 'react';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import './ResultContainer.css';
import CircularProgress from '@material-ui/core/CircularProgress';

const convertDate = (date) => (
  new Date(date).toLocaleDateString()
);

const ListItem = (props) => (
    <li className="listItem">
        <Card>
            <CardContent>
                <Typography className="headline" color="textSecondary" gutterBottom>
                    {props.item.headline.main}
                </Typography>
                <Typography className="source" color="textSecondary" gutterBottom>
                    {`${props.item.source} : ${props.item.document_type} : ${convertDate(props.item.pub_date)}`}
                </Typography>
            </CardContent>
            <CardActions>
                <Button size="small" href={props.item.web_url} target="_blank">Learn More</Button>
            </CardActions>
        </Card>
    </li>
);
  
function list(list) {
    const listItems = list.map((item) =>
        <ListItem
            key={item._id}
            item={item}
        />
    );
    return (
        <ul className="listContainer">
            {listItems}
        </ul>
    );
}

const ResultContainer = props => (
    <>
      {list(props.list)}
      {
        props.isFetchingAgain
        ? <CircularProgress disableShrink />
        : <Button onClick={props.uploadMore}>Show more</Button>
      }
    </>
);

export default ResultContainer;