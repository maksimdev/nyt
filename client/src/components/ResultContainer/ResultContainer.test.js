import React from 'react';
import * as Enzyme from 'enzyme';
import ResultContainer from './ResultContainer';
import Adapter from 'enzyme-adapter-react-16';
import enzymeSerializer from 'enzyme-to-json/serializer';

Enzyme.configure({ adapter: new Adapter() })

expect.addSnapshotSerializer(enzymeSerializer);
describe('ResultContainer', () => {
    const props = {list: []};
    it('should to match snapshot', () => {
        const ResultContainerComponent = Enzyme.shallow(<ResultContainer {...props}/>);
        expect(ResultContainerComponent).toMatchSnapshot();
    });
});