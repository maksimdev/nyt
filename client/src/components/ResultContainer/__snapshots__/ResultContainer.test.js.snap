// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultContainer should to match snapshot 1`] = `
<Fragment>
  <ul
    className="listContainer"
  />
  <WithStyles(Button)>
    Show more
  </WithStyles(Button)>
</Fragment>
`;
